src/main/kotlin/no/nav/syfo/sykmelding/SykmeldingConsumer.kt

```diff
@@ -0,0 +1,90 @@
+package no.nav.syfo.behandler.kafka.sykmelding
+
+import no.nav.syfo.ApplicationState
+import no.nav.syfo.Environment
+import no.nav.syfo.database.DatabaseInterface
+import no.nav.syfo.domain.*
+import no.nav.syfo.kafka.kafkaAivenConsumerConfig
+import no.nav.syfo.personoppgave.createPersonOppgave
+import no.nav.syfo.personoppgave.domain.PersonOppgaveType
+import no.nav.syfo.personoppgave.getPersonOppgaverByReferanseUuid
+import org.apache.kafka.clients.consumer.*
+import java.sql.Connection
+import java.time.*
+import java.util.*
+
+const val SYKMELDING_TOPIC = "teamsykmelding.ok-sykmelding"
+
+fun launchKafkaTaskSykmelding(
+    applicationState: ApplicationState,
+    environment: Environment,
+    database: DatabaseInterface,
+) {
+    val consumerProperties = kafkaAivenConsumerConfig<ReceivedSykmeldingDTO>(environment.kafka).apply {
+        this[ConsumerConfig.AUTO_OFFSET_RESET_CONFIG] = "latest"
+    }
+    val kafkaConsumerSykmelding = KafkaConsumer<String, ReceivedSykmeldingDTO>(consumerProperties)
+
+    kafkaConsumerSykmelding.subscribe(
+        listOf(SYKMELDING_TOPIC)
+    )
+    while (applicationState.ready) {
+        pollAndProcessSykmelding(
+            database = database,
+            kafkaConsumerSykmelding = kafkaConsumerSykmelding,
+        )
+    }
+}
+
+fun pollAndProcessSykmelding(
+    database: DatabaseInterface,
+    kafkaConsumerSykmelding: KafkaConsumer<String, ReceivedSykmeldingDTO>,
+) {
+    val records = kafkaConsumerSykmelding.poll(Duration.ofMillis(1000))
+    if (records.count() > 0) {
+        processSykmelding(
+            database = database,
+            consumerRecords = records,
+        )
+        kafkaConsumerSykmelding.commitSync()
+    }
+}
+
+fun processSykmelding(
+    database: DatabaseInterface,
+    consumerRecords: ConsumerRecords<String, ReceivedSykmeldingDTO>,
+) {
```

Comment:
Vi har vel i det siste prøvd å skille ut dette i en egen service, som snakker med databasen i stedet for at consumeren selv gjør det. Men det går greit for et hackathon tenker jeg 😄